src/cross_toml.rs

```diff
@@ -0,0 +1,194 @@
+#![doc = include_str!("../docs/cross_toml.md")]
+
+use crate::errors::*;
+use crate::Target;
+use serde::Deserialize;
+use std::collections::HashMap;
+
+/// Build environment configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossBuildEnvConfig {
+    volumes: Option<Vec<String>>,
+    passthrough: Option<Vec<String>>,
+}
+
+/// Build configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossBuildConfig {
+    env: Option<CrossBuildEnvConfig>,
+    xargo: Option<bool>,
+    target: Option<String>,
+}
+
+/// Target configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossTargetConfig {
+    passthrough: Option<Vec<String>>,
+    volumes: Option<Vec<String>>,
+    xargo: Option<bool>,
+    image: Option<String>,
+    runner: Option<String>,
+}
+
+/// Wrapper struct for mappings from [`Target`] to [`CrossTargetConfig`]
+///
+/// This is used to circumvent that serde's flatten and field aliases
+/// currently don't work together: <https://github.com/serde-rs/serde/issues/1504>
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossTargetMapConfig {
+    #[serde(flatten)]
+    inner: HashMap<Target, CrossTargetConfig>,
+}
+
+/// Cross configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossToml {
+    #[serde(rename = "target")]
+    targets: Option<CrossTargetMapConfig>,
+    build: Option<CrossBuildConfig>,
```

Comment:
Same here, doesn't need to be option

Reply:
Do you mean the `build` field here? For this I think it makes sense to have the `Option`. For the `targets` field I added the defaulting.

Reply:
my bad, yes I mean the build as well, but if you think it makes sense I'll agree with you :)